app/Dockerfile

```diff
@@ -21,16 +21,21 @@ WORKDIR /usr/src/app
 RUN --mount=type=bind,source=package.json,target=package.json \
     --mount=type=bind,source=package-lock.json,target=package-lock.json \
     --mount=type=cache,target=/root/.npm \
-    npm ci --omit=dev
+    npm ci --include=dev
 
-# Run the application as a non-root user.
-USER node
+# install nodemon for hot-reloading
+RUN npm install -g nodemon
 
 # Copy the rest of the source files into the image.
 COPY . .
 
+
+# Run the application as a non-root user.
+RUN  chown -R node /usr/src/app
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Is this necessary for watch to work properly?

Reply:
yes. `COPY` doesn't manage an earlier `USER` and always make `root` the owner of copied files. Then watch can't update container content relying on `exec` commands